metrics/opencensus_exporter.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2019 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"contrib.go.opencensus.io/exporter/ocagent"
+	"go.opencensus.io/stats/view"
+	"go.uber.org/zap"
+)
+
+func newOpenCensusExporter(config *metricsConfig, logger *zap.SugaredLogger) (view.Exporter, error) {
+	opts := []ocagent.ExporterOption{ocagent.WithServiceName(config.component)}
+	if config.collectorAddress != "" {
+		opts = append(opts, ocagent.WithAddress(config.collectorAddress))
+	}
+	if !config.requireSecure {
```

Comment:
What happens when secure is required?

Reply:
Secure is the default. Unfortunately, it's hard to get TLS set up for many users, so we default here to assuming that the cluster network is "secure enough".

Reply:
I'm assuming you mean the [`WithTLSCredentials`](https://godoc.org/contrib.go.opencensus.io/exporter/ocagent#WithTLSCredentials) option?

I can take a stab at a way to do this, but it's starting to go down a rabbit hole here:

> WithTLSCredentials allows the connection to use TLS credentials when talking to the server. It takes in grpc.TransportCredentials instead of say a Certificate file or a tls.Certificate, because the retrieving these credentials can be done in many ways e.g. plain file, in code tls.Config or by certificate rotation, so it is up to the caller to decide what to use.

I think a secret is probably the best choice, let me see if I can lift some from [`webbook`](https://github.com/knative/pkg/blob/master/webhook/webhook.go#L141).